draft-ietf-quic-multipath.md

```diff
@@ -478,10 +478,12 @@ The receiver of a PATH_ABANDON frame MAY also send
 a PATH_ABANDON frame to indicate its own unwillingness to receive
 any packet on this path anymore.
 
-The PATH_ABANDON frame retires the associated Path ID.
-When an endpoint receives a PATH_ABANDON frame,
+Receiption of the PATH_ABANDON frame is the first step to retire a
+path ID. However, the Path ID can only be retired after all active
+connection IDs for the Path ID have been retired.
```

Comment:
I'm considering that there might be situations that attackers doesn't send CID retirement, on purpose to keep the path id and path resources not to be freed, and it could cause memory increase under this situation. 

Would it be a better way that we use Path Abandon like Connection_Close in RFC9000? After reception of Path Abandon, Endpoints could still send ACK_MP and MP_CID_Retirements within 3 PTO, but if endpoints doesn't send CID retirement or the packets containing CID retirements just lost, the peer could still be able to close the Path and release all the resources related after a period of time.

Reply:
This is not a problem because Maximum Paths limits the number of open paths. So if an attacker does not retire all CIDs, the path gets not closed but it can also not open more path when the Maximum Paths limit is reached.

Reply:
The Maximum Paths is not a (relative) counter indicating up to how many paths can be concurrently open. It indicates the maximum Path ID (or a way to derive the maximum path ID if we consider even/odd split) that can be used by the peer over the connection.

Regarding the first part about resource release, I agree that we should enable endpoints to release the resources related to the to-be-closed path if it has not seen CID retirement in the 3 PTO interval. This would make the endpoint ignores ("drops") further packets belonging to that path, which is in the end the idea of a closed path.

Reply:
Should we maybe say "release all ressources related to the Path ID" instead of "retire the Path ID"?

So what happens is, if you send an Abandon frame, you wait 3 PTOs, then you send the Retire CID frames of your connection IDs and release them. Further, if you can also release the CID that you got from your peer for this path ID, even if you didn't receive a retire CID frame from your peer, as you will anyway not use them anymore. The only risk is that the peer believes the path is not retired and therefore doesn't issue new CIDs for a potential new path if the path limit is reached. However, there is nothing you can do about this (as long as the abandon frame you've sent was acknowledged by the peer).

So we need to be more clear about this?